orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/config/CloudDriverConfiguration.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.clouddriver.*;
+import com.netflix.spinnaker.orca.jackson.OrcaObjectMapper;
+import com.netflix.spinnaker.orca.reactive.RequestMetricsProfiler;
+import com.netflix.spinnaker.orca.reactive.WindowedRequestMetrics;
+import com.netflix.spinnaker.orca.reactive.WritableRequestMetrics;
+import com.netflix.spinnaker.orca.retrofit.RetrofitConfiguration;
+import com.netflix.spinnaker.orca.retrofit.logging.RetrofitSlf4jLog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import retrofit.Profiler;
+import retrofit.RequestInterceptor;
+import retrofit.RestAdapter;
+import retrofit.client.Client;
+import retrofit.converter.JacksonConverter;
+
+import java.lang.reflect.InvocationTargetException;
+import java.time.Duration;
+
+import static java.util.stream.Collectors.toList;
+import static retrofit.Endpoints.newFixedEndpoint;
+
+@Configuration
+@Import(RetrofitConfiguration.class)
+@ComponentScan({
+  "com.netflix.spinnaker.orca.clouddriver",
+  "com.netflix.spinnaker.orca.kato.pipeline",
+  "com.netflix.spinnaker.orca.kato.tasks"
+})
+@EnableConfigurationProperties(CloudDriverConfigurationProperties.class)
+class CloudDriverConfiguration {
+
+  @ConditionalOnMissingBean(ObjectMapper.class)
+  @Bean
+  ObjectMapper mapper() {
+    return OrcaObjectMapper.newInstance();
+  }
+
+  @Bean
+  WritableRequestMetrics cloudDriverRequestMetrics(
+    @Value("${cloudDriver.qos.metrics.window.seconds:300}")
+      long qosMetricsWindowSeconds) {
+    return new WindowedRequestMetrics("CloudDriver", Duration.ofSeconds(qosMetricsWindowSeconds));
+  }
```

Comment:
New bean that can track metrics for Retrofit requests

Reply:
Yeah, @robzienert was against using Atlas which was what I'd assumed would make sense. Maybe he can chime in on what the justification was.

Reply:
One thing I'd say is I'd like us to get ahead of alerts here. i.e. start throttling _before_ alerts start firing.

Reply:
I'd keep Atlas for alerting on errors, I think it makes sense to have built-in logic for throttling

Reply:
I'm not completely sold